src/number_format.js

```diff
@@ -591,12 +591,16 @@ class NumberFormat extends React.Component {
      or if value is empty string (when whole input is cleared)
      or whole input is replace with a number
     */
+    const leftBound = !!format ? 0 : prefix.length;
+    const rightBound = lastValue.length - (!!format ? 0 : suffix.length);
     if (
       value.length > lastValue.length
       || !value.length ||
       start === end ||
       (start === 0 && end === lastValue.length) ||
```

Comment:
logically when ever `start === 0 && end === lastValue.length` is true `selectionStart === 0 && selectionEnd === lastValue.length` will always be true. Same for leftBound and rightBound condition. 
I will remove the extra condition.

Reply:
For understanding more about these conditions, 
- For selection condition - PR: #516, Issue: #515 
- For start and end condition - PR: #511, Issue: #450